src/rules/consecutiveFunctionOverloadsRule.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "consecutive-function-overloads",
+        description: "Enforces function overloads to be consecutive.",
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: ["true"],
+        type: "style",
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_STRING_FACTORY = (name: string) => `All '${name}' signatures should be adjacent`;
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoNonConsecutiveFunctionOverload(sourceFile, this.getOptions()));
+    }
+}
+
+function isStringOrNumericLiteral(kind: ts.SyntaxKind) {
+    return kind === ts.SyntaxKind.StringLiteral || kind === ts.SyntaxKind.NumericLiteral;
+}
+
+function getTextOfPropertyName(name: ts.PropertyName): string {
```

Comment:
if you create the type guard I suggest above, this can be a little simpler:

```ts
switch (name.kind) {
    case ts.SyntaxKind.Identifier:
        return (name as ts.Identifier).text;
    case ts.SyntaxKind.ComputedPropertyName:
        const { expression } = (name as ts.ComputedPropertyName);
        if (isLiteralExpression(expression)) {
            return expression.text;
        }
     default:
        if (isLiteralExpression(name)) {
            return name.text;
        }
}
```